Objects.Inheritance.py

```diff
@@ -0,0 +1,136 @@
+'''
+Creation of a class "figure" with properties: coordinates of 4 points. The class contains methods
+for checking the existence of the created quadrilateral, calculation and output of information
+about a figure: lengths of sides, diagonals, perimeter, area of a figure.
+The class has a check of the figure for convexity.
+Added derived class "parallelogram". The class provides a check whether the figure is a parallelogram.
+
+Below is a commented out program that analyzes quadrilaterals to find the maximum area among objects.
+'''
+import math
+
+
+class Figure:
+    def __init__(self, x1, y1, x2, y2, x3, y3, x4, y4):
```

Comment:
Pairs of the coordinates could be wrapped to the additional class `Point`.
After that, all those points could be stored in `Figure` as a list.

Reply:
In my opinion, creating a new class will require more memory. I've created a list in the "Figure" class that stores the coordinates of each point as a lists. Point coordinates are set sequentially and manually by the user